### DISCUSSION TITLE
Open Source SMO

### DISCUSSION THREAD

**Discussion by jzabroski** [microsoft/sqlmanagementobjects#42] [Ideas]
Hi,

I would be interested in fixing some bugs if SMO were to become open source, as well as make some API improvements to take advantage of C# 9 pattern matching.

Is this possible now?

**Comment by shueybubbles** [comment]
It's on my to-do list. There's a lot of overhead involved, mainly cleaning out references to internal Microsoft resources and adding copyright/license verbiage to all the files etc. I don't have an ETA, unfortunately.

**Comment by dzsquared** [comment]
@jzabroski  - good news  😄 
https://github.com/microsoft/sqlmanagementobjects/pull/82

**Comment by shueybubbles** [comment]
I've pushed a snapshot of the SMO code from which nuget package version 161.47008.0 was built.
We will continue to update the public repo when we ship the nuget package.
We are willing to accept public pull requests too. I have a fair amount of work to do to document requirements for PRs, though.
